pyscf/lno/lnoccsd.py

```diff
@@ -0,0 +1,923 @@
+        nfrag = len(frag_res)
+        efrag_pt2 = np.zeros(nfrag)
+        efrag_cc = np.zeros(nfrag)
+        efrag_cc_t = np.zeros(nfrag)
+        efrag_pt2_spin_comp = np.zeros((nfrag,2))
+        efrag_cc_spin_comp = np.zeros((nfrag,2))
+
+        for i in range(nfrag):
+            ept2, ecc, ecc_t = frag_res[i]
+            efrag_pt2[i] = float(ept2)
+            efrag_cc[i] = float(ecc)
+            efrag_cc_t[i] = float(ecc_t)
+            efrag_pt2_spin_comp[i] = ept2.spin_comp
+            efrag_cc_spin_comp[i] = ecc.spin_comp
+        self.efrag_pt2  = efrag_pt2  * frag_wghtlist
+        self.efrag_cc   = efrag_cc   * frag_wghtlist
+        self.efrag_cc_t = efrag_cc_t * frag_wghtlist
+        self.efrag_pt2_spin_comp  = efrag_pt2_spin_comp  * frag_wghtlist[:,None]
+        self.efrag_cc_spin_comp   = efrag_cc_spin_comp   * frag_wghtlist[:,None]
+
+    def _finalize(self):
+        r''' Hook for dumping results and clearing up the object.'''
+        logger.note(self, 'E(%s) = %.15g  E_corr = %.15g',
+                    'LNOMP2', self.e_tot_pt2, self.e_corr_pt2)
+        logger.note(self, 'E(%s) = %.15g  E_corr = %.15g',
+                    'LNOCCSD', self.e_tot, self.e_corr)
+        if self.ccsd_t:
+            logger.note(self, 'E(%s) = %.15g  E_corr = %.15g',
+                        'LNOCCSD_T', self.e_tot_ccsd_t, self.e_corr_ccsd_t)
+        logger.note(self, 'Summary by spin components')
+        logger.note(self, 'LNOMP2   Ess = %.15g  Eos = %.15g  Escs = %.15g',
+                    self.e_corr_pt2_ss, self.e_corr_pt2_os, self.e_corr_pt2_scs)
+        logger.note(self, 'LNOCCSD  Ess = %.15g  Eos = %.15g  Escs = %.15g',
+                    self.e_corr_ccsd_ss, self.e_corr_ccsd_os, self.e_corr_ccsd_scs)
+        return self
+
+    @property
+    def e_tot_scf(self):
+        return self._scf.e_tot
+
+    @property
+    def e_corr(self):
+        return self.e_corr_ccsd
+
+    @property
+    def e_tot(self):
+        return self.e_corr + self.e_tot_scf
+
+    @property
+    def e_corr_ccsd(self):
+        e_corr = np.sum(self.efrag_cc)
+        return e_corr
+
+    @property
+    def e_corr_ccsd_ss(self):
+        e_corr = np.sum(self.efrag_cc_spin_comp[:,0])
+        return e_corr
+
+    @property
+    def e_corr_ccsd_os(self):
+        e_corr = np.sum(self.efrag_cc_spin_comp[:,1])
+        return e_corr
+
+    @property
+    def e_corr_ccsd_scs(self):
+        e_corr = 0.333*self.e_corr_ccsd_ss + 1.2*self.e_corr_ccsd_os
+        return e_corr
+
+    @property
+    def e_corr_pt2(self):
+        e_corr = np.sum(self.efrag_pt2)
+        return e_corr
+
+    @property
+    def e_corr_pt2_ss(self):
+        e_corr = np.sum(self.efrag_pt2_spin_comp[:,0])
+        return e_corr
+
+    @property
+    def e_corr_pt2_os(self):
+        e_corr = np.sum(self.efrag_pt2_spin_comp[:,1])
+        return e_corr
+
+    @property
+    def e_corr_pt2_scs(self):
+        e_corr = 0.333*self.e_corr_pt2_ss + 1.2*self.e_corr_pt2_os
```

Comment:
maybe the ss, os factor can be configurable arguments, like the DF SCS-MP2 does

https://github.com/pyscf/pyscf/blob/e155c487f6f44dc249089c7d6aafded58ce53e0e/pyscf/mp/dfmp2_native.py#L138

https://github.com/pyscf/pyscf/blob/e155c487f6f44dc249089c7d6aafded58ce53e0e/pyscf/mp/dfmp2_native.py#L262